lib/vtls/gtls.c

```diff
@@ -344,8 +344,8 @@ static CURLcode handshake(struct Curl_cfilter *cf,
       return CURLE_SSL_CONNECT_ERROR;
     }
 
-    /* Reset our connect state machine */
-    connssl->connecting_state = ssl_connect_1;
+    /* Advance our connect state machine */
+    connssl->connecting_state = ssl_connect_3;
```

Comment:
Is this not odd that you can just change this unconditionally even if early data is not requested?

Reply:
This is more of a refactor so that the code looks more like what is happening in all the other backends
`ssl_connect_1` -> Setting up context, and other bookkeeping
`ssl_connect_2` -> Actual Handshake
`ssl_connect_3` -> Verifying peers

Previously `ssl_connect_3` was not utilised, `handshake()` completed the handshake and in `gtls_connect_common` the verification part followed unconditionally.